trainer/distribute.py

```diff
@@ -51,7 +52,10 @@ def distribute():
         command[-1] = f"--rank={rank}"
         # prevent stdout for processes with rank != 0
         stdout = None
-        p = subprocess.Popen(["python3"] + command, stdout=stdout, env=my_env)  # pylint: disable=consider-using-with
+        if sys.platform == 'win32':
+            p = subprocess.Popen(["python"] + command, stdout=stdout, env=my_env)  # pylint: disable=consider-using-with
+        else:
+            p = subprocess.Popen(["python3"] + command, stdout=stdout, env=my_env)  # pylint: disable=consider-using-with
```

Comment:
The correct, platform-and-virtualenv-agnostic way to do this is `sys.executable`.

```suggestion
        p = subprocess.Popen([sys.executable, *command], stdout=stdout, env=my_env)  # pylint: disable=consider-using-with
```

Reply:
I guess we don't need to run this if it is not Win32.